tests/tests_full_integration_dyndns.yml

```
@@ -0,0 +1,179 @@
+
+# Example inventory:
+# [client]
+# client1 ansible_host=<CLIENT IP>\
+#  ad_integration_password=Secret123 ad_integration_realm="domain.com" \
+#  ad_dyndns_update=true
+# [ad]
+# ad1 ansible_host=<AD IP> ansible_connection=winrm ansible_password=Secret123\
+#  ansible_port=5986 ansible_user=Administrator \
+#  ansible_winrm_server_cert_validation=ignore
+
+- name: Set variables
+  hosts: client
+  tasks:
+    - name: Set ad_integration_realm if it is not defined or is empty
+      set_fact:
+        ad_integration_realm: domain.com
+      when: ad_integration_realm is not defined or not ad_integration_realm
+
+- name: Configure DynDNS on AD
+  hosts: ad
+  gather_facts: true
+  tasks:
+    - name: Get AD address
+      set_fact:
+        ad_address: "{{ hostvars[inventory_hostname].ansible_host }}"
+    - name: Get network for zone
+      set_fact:
+        network_ad: >-
+          "{{ ad_address.split('.')[-2::-1] | join('.') }}.in-addr.arpa"
+    - name: show network
+      debug:
+        msg: "AD network: {{ network_ad }}"
+    - name: Create reverse zone
+      win_command: dnscmd.exe /zoneadd {{ network_ad }} /primary
+      register: res
+      failed_when:
+      - 'res.rc != 0 and "DNS_ERROR_ZONE_ALREADY_EXISTS" not in res.stdout'
+    - name: Allow updates in the zone
+      win_command: dnscmd.exe /config {{ hostvars[groups['client'][0]].
+        ad_integration_realm }} /allowupdate 1
+    - name: Allow updates in the reverse zone
+      win_command: dnscmd.exe /config {{ network_ad }} /allowupdate 1
+    - name: Disable dns forwarders
+      win_command: dnscmd.exe /config /norecursion 1
+
+- name: Ensure that the role configures dynamic dns
+  hosts: client
+  tasks:
+    - name: Run the integration
+      block:
+        - name: Debug dyndns settings
+          debug:
+            msg: >-
+              "Interface {{ _ad_dyndns_iface | default('eth0') }} and server
+              : {{ hostvars[groups['ad'][0]].ansible_host }}"
+        - name: Run the system role with proper config
+          include_role:
+            name: linux-system-roles.ad_integration
+          vars:
+            ad_dyndns_server: "{{ hostvars[groups['ad'][0]].ansible_host }}"
+            ad_dyndns_iface: "{{ _ad_dyndns_iface | default('eth0') }}"
+            ad_dyndns_auth: "none"
+            ad_dyndns_update: true
+            ad_dyndns_refresh_interval: 60
+        - name: Apply additional changes on sssd
+          ini_file:
+            path: /etc/sssd/sssd.conf
+            state: present
+            section: "domain/{{ ad_integration_realm | lower }}"
+            option: "{{ item.key }}"
+            value: "{{ item.value }}"
+            create: true
+          loop:
+          - key: debug_level
+            value: "{{ 9 | int }}"
+        - name: Clean sssd log
+          command: >-
+            'truncate -s 0  /var/log/sssd/sssd_{{ ad_integration_realm }}.log'
+        - name: Restart sssd
+          command: systemctl restart sssd
```

```suggestion
          service:
            name: sssd
            status: restarted
```
Ansible strongly prefers the use of modules instead of `command` and `shell` - not sure why ansible-lint didn't complain about this unless the issue was masked by the other issues in the file